.github/workflows/main.yml

```diff
@@ -0,0 +1,40 @@
+name: CI Pipeline
+
+on:
+    pull_request:
+        branches:
+            - '**'
+    push:
+        branches:
+            - '**'
```

Comment:
For both, `push` and `pull_requests`, the workflow should run only on the `main` branch.

Reply:
Done!